.changeset/wet-trainers-heal.md

```diff
@@ -0,0 +1,5 @@
+---
+'@itwin/itwinui-react': minor
+---
+
+Added dependency on `@tanstack/react-virtual`.
```

Comment:
I think we should test out that this library works fine for `Table` and `Tree` cases before we finalize it. It would be problematic if we add this dependency and make a release, then later find out that it doesn't handle everything we need.

So while it's fine for this PR to be its own separate thing, the `Table`/`Tree` parts need to be investigated sooner rather than later.

Reply:
I do have branches testing this library with the `Table` and `Tree` components. There were some keyboard navigation issues with the `Tree` component, and the `Table` component started failing multiple unit tests because of the added state hook for holing the `table` element. I believe these issues should be fixable, but I will finish investigating them before merging this PR.

Reply:
I have created draft PRs for both the [`Table`](https://github.com/iTwin/iTwinUI/pull/2092) and [`Tree`](https://github.com/iTwin/iTwinUI/pull/2091) components. They should be far enough along at this point to show that the `tanstack` virtual package will work for these components as well.